src/ucm/util/reloc.c

```diff
@@ -192,8 +197,17 @@ ucm_reloc_modify_got(ElfW(Addr) base, const ElfW(Phdr) *phdr, const char *phname
                 ucm_error("failed to modify GOT page %p to rw: %m", page);
                 return UCS_ERR_UNSUPPORTED;
             }
-            patch->prev_value = *entry;
-            *entry = patch->value;
+            success = dladdr(*entry, &entry_dlinfo);
+            ucs_assertv_always(success, "can't find shared object with entry %p",
+                               *entry);
+
+            /* store default entry to prev_value to guarantee valid pointers
+             * throughout life time of the process */
+            if (ctx->def_dlinfo.dli_fbase == entry_dlinfo.dli_fbase) {
+                ctx->patch->prev_value = *entry;
+            }
```

Comment:
let's update the debug message ucm_trace from line 191 - to be printed only if we modify, and add another ucm_trace of 'else' case if we decided not to modify

Reply:
we always modify, will add a trace for storing origin